server/pulp/server/agent/direct/pulpagent.py

```diff
@@ -111,14 +114,15 @@ def unregistered(context):
         :param context: The call context.
         :type context: pulp.server.agent.context.Context
         """
-        agent = Agent(
-            context.url,
-            context.route,
-            secret=context.secret,
-            authenticator=context.authenticator,
-            wait=0)
-        consumer = agent.Consumer()
-        consumer.unregistered()
+        with context:
+            agent = Agent(
+                context.url,
+                context.address,
+                secret=context.secret,
+                authenticator=context.authenticator,
+                wait=0)
+            consumer = agent.Consumer()
+            consumer.unregistered()
```

Comment:
Should this function be renamed to `unregister` instead of `unregistered`? unregistered sounds like it returns state, but this function doesnt' return state it cases a state change so it should be a verb.